build.psm1

```diff
@@ -2997,6 +2997,12 @@ function New-TestPackage
         Write-Verbose -Message "Creating destination folder: $Destination"
     }
 
+    $rootFolder = $env:TEMP
+
+    if (-not $rootFolder -and $env:TF_BUILD) {
+        $rootFolder = $env:AGENT_WORKFOLDER
+    }
```

Comment:
should there be an `else` here, just in case?

Reply:
First it tries to set root folder to $env:temp if it is $null then it tries to set it to AGENT_WOORKFOLDER only if running under VSTS (checked by $env:TF_BUILD)